pytorch_lightning/trainer/connectors/logger_connector/result.py

```diff
@@ -575,16 +562,16 @@ def any_tensor(_):
 
             # populate logging metrics
             if result_metric.meta.logger:
-                metrics[MetricSource.LOG][forked_name] = value
+                metrics["log"][forked_name] = value
```

Comment:
Let's use self.CALLBACK instead from the ResultCollection. Same for log, pbar.

Reply:
That's not necessary now that it's fully typed, see: https://github.com/PyTorchLightning/pytorch-lightning/pull/9270#discussion_r700989735

PyCharm reports the error:

<img width="465" alt="Screenshot 2021-09-02 at 13 26 27" src="https://user-images.githubusercontent.com/11067892/131835992-620ff694-d584-4037-8b15-ee7dc7c1d82a.png">

and mypy:

```python
pytorch_lightning/trainer/connectors/logger_connector/result.py:569: error: TypedDict "_METRICS" has no key "calback"  [typeddict-item]
pytorch_lightning/trainer/connectors/logger_connector/result.py:569: note: Did you mean "callback"?
```